tests/ui/needless_match.rs

```diff
@@ -103,6 +103,23 @@ fn if_let_option() {
     } else {
         None
     };
+
+    // Don't trigger
+    let _ = if let Some(a) = Some(1) {
+        do_something();
```

Comment:
Though actually one small nit, the `do_something` should go as this statement doesn't trigger the FP

```suggestion
```

Reply:
@Alexendoo thank you! I fixed it